packages/api-plugin-fulfillment/migrations/2.js

```diff
@@ -0,0 +1,177 @@
+  "reaction:legacy:fulfillmentTypes/read",
+  "reaction:legacy:fulfillmentTypes/update",
+  "reaction:legacy:fulfillmentMethods/create",
+  "reaction:legacy:fulfillmentMethods/delete",
+  "reaction:legacy:fulfillmentMethods/read",
+  "reaction:legacy:fulfillmentMethods/update"
+];
+
+/**
+ * @summary migrates the database down one version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function down({ db, progress }) {
+  progress(0);
+
+  const allGroups = await db.collection("Groups").find({}).toArray();
+  const affectedGlobalGroups = [];
+  allGroups.forEach((group) => {
+    const perms = group.permissions;
+    if (perms && Array.isArray(perms) && perms.length) {
+      const found = newGlobalPermissions.some((elem) => perms.includes(elem));
+      if (found) affectedGlobalGroups.push(group.slug);
+    }
+  });
+
+  await db.collection("Groups").updateMany({
+    slug: { $in: affectedGlobalGroups }
+  }, {
+    $pullAll: { permissions: newGlobalPermissions }
+  });
+  progress(10);
+
+  await db.collection(COLL_FF_DEST).drop();
+  progress(50);
+
+  await db.collection(COLL_FFR_DEST).drop();
+  progress(100);
+}
+
+/**
+ * @summary Performs migration up from previous data version
+ * @param {Object} context Migration context
+ * @param {Object} context.db MongoDB `Db` instance
+ * @param {Function} context.progress A function to report progress, takes percent
+ *   number as argument.
+ * @return {undefined}
+ */
+async function up({ db, progress }) {
+  progress(0);
+
+  const shippingCopyResp = await db.collection(COLL_FF_SOURCE).aggregate([{ $match: {} }, { $out: COLL_FF_DEST }]).next();
+  if (shippingCopyResp) throw new Error("Error in copying Shipping collection"); // above command returns null if successful
+  progress(20);
+
+  const flatRateRestCopyResp = await db.collection(COLL_FFR_SOURCE).aggregate([{ $match: {} }, { $out: COLL_FFR_DEST }]).next();
+  if (flatRateRestCopyResp) throw new Error("Error in copying FlatRateFulfillmentRestrictions collection"); // above command returns null if successful
+  progress(40);
+
+  const operations = [];
+
+  const ffTypeUpdate = {
+    updateMany: {
+      filter: { fulfillmentType: { $exists: false } },
+      update: {
+        $set: {
+          fulfillmentType: FF_TYPE
+        }
+      }
+    }
+  };
+
+  const ffMethodUpdate = {
+    updateMany: {
+      filter: { methods: { $exists: true } },
+      update: {
+        $set: {
+          "methods.$[eachMethod].fulfillmentMethod": FF_METHOD
+        }
+      },
+      arrayFilters: [
+        {
+          "eachMethod.fulfillmentMethod": { $exists: false }
+        }
+      ]
+    }
+  };
+
+  operations.push(ffTypeUpdate);
+  operations.push(ffMethodUpdate);
+
+  const bulkWriteResp = await db.collection(COLL_DEST).bulkWrite(operations);
```

Comment:
This will probably work but it's not ready what the bulkWrite API is really for an would probably be better as written as a single query. Bulkwrite is when you are doing a large amount of atomic operations, not one large one.